python/pyarrow/tests/test_substrait.py

```diff
@@ -165,3 +169,108 @@ def test_get_supported_functions():
+
+
+@pytest.mark.skipif(sys.platform == 'win32',
+                    reason="ARROW-16392: file based URI is" +
+                    " not fully supported for Windows")
+def test_run_aggregate_query(tmpdir):
+    substrait_query = """
+    {
+        "relations": [{
+        "rel": {
+            "aggregate": {
+            "input": {
+                "read": {
+                "base_schema": {
+                    "names": ["A", "B", "C"],
+                    "struct": {
+                    "types": [{
+                        "i64": {}
+                    }, {
+                        "i64": {}
+                    }, {
+                        "i64": {}
+                    }]
+                    }
+                },
+                "local_files": {
+                    "items": [
+                    {
+                        "uri_file": "file://FILENAME_PLACEHOLDER",
+                        "arrow": {}
+                    }
+                    ]
+                }
+                }
+            },
+            "groupings": [{
+                "groupingExpressions": [{
+                "selection": {
+                    "directReference": {
+                    "structField": {
+                        "field": 0
+                    }
+                    }
+                }
+                }]
+            }],
+            "measures": [{
+                "measure": {
+                "functionReference": 0,
+                "arguments": [{
+                    "value": {
+                    "selection": {
+                        "directReference": {
+                        "structField": {
+                            "field": 1
+                        }
+                        }
+                    }
+                    }
+                }],
+                "sorts": [],
+                "phase": "AGGREGATION_PHASE_INITIAL_TO_RESULT",
+                "invocation": "AGGREGATION_INVOCATION_ALL",
+                "outputType": {
+                    "i64": {}
+                }
+                }
+            }]
+            }
+        }
+        }],
+        "extensionUris": [{
+        "extension_uri_anchor": 0,
+        "uri": "AGGREGATE_URI_PLACEHOLDER"
+        }],
+        "extensions": [{
+        "extension_function": {
+            "extension_uri_reference": 0,
+            "function_anchor": 0,
+            "name": "count"
+        }
+        }]
+    }
+    """
+
+    file_name = "read_agg_data.arrow"
+    table = pa.Table.from_arrays([[1, 2, 3, 2, 3, 1, 1],
+                                  [10, 20, 30, 40, 50, 60, 70],
+                                  [3, 4, 5, 1, 2, 0, 20]],
+                                 names=['A', 'B', 'C'])
+    path = _write_dummy_data_to_disk(tmpdir, file_name, table)
+
+    query_with_path = substrait_query.replace("FILENAME_PLACEHOLDER", path)
+    query_with_ext_uri = query_with_path.replace(
+        "AGGREGATE_URI_PLACEHOLDER", _kAggregateURI)
```

Comment:
Are we going to need to do this for all of our substrait plans?  Shouldn't Acero be handling this part?

Reply:
This is part of the Substrait query, right? You would stil have to reference it AFAIK.

Reply:
The substrait plan is generated with the name of the yaml file in it already.  I think @westonpace is actually going to submit a PR to automatically reference the proper URI given the name of the yaml.

Reply:
https://issues.apache.org/jira/browse/ARROW-17736 explains what I'm thinking.

Reply:
I will work on this one.

Reply:
Ah.  I might keep this one actually.  I already have a PR ready to go from a few weeks back when I was playing around with this.  I just didn't have time to submit it, which I will do now.  I would appreciate a review / opinions on the change however.